server/superdesk/io/rfc822.py

```diff
@@ -0,0 +1,147 @@
+                        body = part.get_payload(decode=True)
+                        text_body = body.decode('utf-8')
+                        continue
+                    if part.get_content_type() == "text/html":
+                        body = part.get_payload(decode=True)
+                        html_body = body.decode('utf-8')
+                        continue
+                    if part.get_content_maintype() == 'multipart':
+                        continue
+                    if part.get('Content-Disposition') is None:
+                        continue
+                    # we are only going to pull off image attachments at this stage
+                    if part.get_content_maintype() != 'image':
+                        continue
+
+                    fileName = part.get_filename()
+                    if bool(fileName):
+                        image = part.get_payload(decode=True)
+                        content = io.BytesIO(image)
+                        res = process_file_from_stream(content, part.get_content_type())
+                        file_name, content_type, metadata = res
+                        content.seek(0)
+                        image_id = self.parser_app.media.put(content, filename=fileName,
+                                                             content_type=content_type, metadata=metadata)
+                        renditions = {'baseImage': {'href': image_id}}
+
+                        media_item = dict()
+                        media_item['guid'] = generate_guid(type=GUID_TAG)
+                        media_item['type'] = 'picture'
+                        media_item['renditions'] = renditions
+                        media_item['mimetype'] = content_type
+                        media_item['filemeta'] = metadata
+                        media_item['slugline'] = fileName
+                        if text_body is not None:
+                            media_item['body_html'] = text_body
+                        media_item['headline'] = item['headline']
+                        new_items.append(media_item)
+
+                        # add a reference to this item in the composite item
+                        media_ref = {}
+                        media_ref['guid'] = media_item['guid']
+                        media_ref['residRef'] = media_item['guid']
+                        media_ref['headline'] = fileName
+                        media_ref['location'] = 'ingest'
+                        media_ref['itemClass'] = 'icls:picture'
+                        refs.append(media_ref)
+                        groups = []
+                        comp_item['groups'] = groups
+
+        if html_body is not None:
+            item['body_html'] = html_body
+        else:
+            item['body_html'] = text_body
+            item['type'] = 'preformatted'
+
+        # if there is more than 1 ref then it because we added an image
+        if len(refs) > 1:
```

Comment:
It is not a test for an empty list,  "> 1" is good, as it would have a single entry for the text item if it had no image. Any way have re-factored it in the hope it is clearer.